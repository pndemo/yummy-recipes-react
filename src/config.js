/*
* Application configurations
*/

import axios from 'axios';

// API URLs
export const baseAPIURL = 'http://yummy-recipes-apis.herokuapp.com';
export const authAPIURL = baseAPIURL.concat('/api/v1/auth/');
export const categoryAPIURL = baseAPIURL.concat('/api/v1/category/');
export const recipeAPIURL = baseAPIURL.concat('/api/v1/recipe/');

// axios instance for private API calls
const privateAxiosInstance = axios.create({
  headers: { Authorization: `Bearer ${localStorage.getItem('token')}` },
});
privateAxiosInstance.interceptors.request.use((config) => {
  if (localStorage.getItem('token') && config.headers.Authorization === 'Bearer null') {
    config.headers.Authorization = `Bearer ${localStorage.getItem('token')}`;
  }
  return config;
});

export default privateAxiosInstance;
