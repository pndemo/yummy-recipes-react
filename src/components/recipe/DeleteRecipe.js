/*
* -- Delete recipe application component --
*  Enables a user to delete a recipe
*/

import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { recipeAPIURL } from '../../config';
import privateAxiosInstance from '../../config';
import Header from '../common/Header';
import FooterFlex from '../common/FooterFlex';
import axios from 'axios'
import Time from 'react-time-format'
import { ToastContainer, toast } from 'react-toastify';

// Application styling
import '../../static/App.css';

class DeleteRecipe extends Component {
    // App component properties
    constructor(props) {
        super(props);
        this.state = {
            isDeleted: false,
            recipeName: '',
        };
    }

    // Get recipe details
    componentWillMount() {
        privateAxiosInstance.get(`${ recipeAPIURL }${ this.props.match.params.category_id }/${ this.props.match.params.recipe_id }`)
        .then((response) => {
            if (response.status === 200) {
                this.setState({
                    recipeName: response.data['recipe_name'],
                });
            }
        })
        .catch((error) => {
            if (error.response){
                document.getElementById("error").innerHTML = error.response.data['message'];
            }
        });
    }

    // Handle delete recipe
    deleteRecipeHandler = (event) => {
        event.preventDefault();
        privateAxiosInstance.delete(`${ recipeAPIURL }${ this.props.match.params.category_id }/${ this.props.match.params.recipe_id }`,
        {
            recipe_name: this.state.recipeName,
        })
        .then((response) => {
            this.setState({
                isDeleted: true,
            })
            toast.success('Recipe has been deleted.')
        })
        .catch((error) => {
            if (error.response){
                document.getElementById("error").innerHTML = error.response.data['message'];
            }
        });
    };

    // Component rendering
    render() {
        if (this.state.isDeleted) {
            return <Redirect to={ '/recipes/'+this.props.match.params.category_id }/>
        }
        return (
            <div className="App-light">
                <Header/>
                <div className="container">
                    <div className="row">
                        <div className="col-xs-12">
                            <h1 className="App-page-title">Delete Recipe</h1>
                        </div>
                    </div>
                    <div className="row">
                        <div className="col-xs-12 col-md-8 col-lg-6 col-md-offset-2 col-lg-offset-3">
                            <p>Do you want to delete recipe
                                <strong> { this.state.recipeName }</strong>?
                            </p>
                            <form onSubmit={ this.deleteRecipeHandler }>
                                <p id="error" className="error"></p>
                                <div className="form-group">
                                    <button type="submit" className="btn btn-primary btn-block App-btn-add">Delete</button>
                                </div>
                            </form>
                        </div>
                    </div>
                </div>
                <FooterFlex/>
            </div>
        );
    }
}

export default DeleteRecipe;
