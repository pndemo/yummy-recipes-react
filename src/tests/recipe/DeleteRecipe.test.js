import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import DeleteRecipe from '../../components/recipe/DeleteRecipe';
import { recipeAPIURL } from '../../config';

jest.mock('react-notifications');

const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

const mock = new MockAdapter(axios);

mock.onDelete(`${recipeAPIURL}1/1`).reply(200, {});

describe('Component: DeleteRecipe', () => {
  const deleteRecipeMock = jest.fn();
  const deleteRecipeComponent = mount(<DeleteRecipe match={{ params: { category_id: 1, recipe_id: 1 } }} DeleteRecipe={deleteRecipeMock} />);
  const deleteRecipeButton = deleteRecipeComponent.find('[type="submit"]');

  it('Display DeleteRecipe component', () => {
    const rendered = renderer.create(
      <DeleteRecipe match={{ params: { category_id: 1, recipe_id: 1 } }} />,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('Delete a recipe', () => {
    deleteRecipeButton.simulate('submit');
    expect(deleteRecipeMock.mock.calls.length === 1);
  });
});
