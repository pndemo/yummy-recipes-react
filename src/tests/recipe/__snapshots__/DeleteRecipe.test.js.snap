// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: DeleteRecipe Display DeleteRecipe component 1`] = `
<div
  className="App-light"
>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <button
          aria-expanded="false"
          className="navbar-toggle collapsed nav-button"
          data-target="#navigationbar"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
        <a
          className="App-title-navbar navbar-brand"
          href="/"
        >
          <span
            className="glyphicon glyphicon-fire"
          />
           Yummy Recipes
        </a>
      </div>
      <div
        className="collapse navbar-collapse"
        id="navigationbar"
      >
        <ul
          className="nav navbar-nav navbar-right"
        >
          <li>
            <a
              href="/change_password"
            >
              <span
                className="glyphicon glyphicon-user"
              />
               Change Password
            </a>
          </li>
          <li>
            <a
              href="/logout"
            >
              <span
                className="glyphicon glyphicon-log-out"
              />
               Sign out
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <h1
          className="App-page-title"
        >
          Delete Recipe
        </h1>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-md-8 col-lg-6 col-md-offset-2 col-lg-offset-3"
      >
        <p>
          Do you want to delete recipe
          <strong>
             
            
          </strong>
          ?
        </p>
        <form
          onSubmit={[Function]}
        >
          <p
            className="error"
            id="error"
          />
          <div
            className="form-group"
          >
            <button
              className="btn btn-primary btn-block App-btn-add"
              type="submit"
            >
              Delete
            </button>
            <a
              className="btn btn-primary btn-block App-btn-cancel"
              href="/recipes/1"
              role="button"
            >
              CANCEL
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>
  <footer
    className="App-footer-flex"
  >
    <p
      className="App-copyright"
    >
      Â© 2018 Yummy Recipes LLC
    </p>
  </footer>
</div>
`;
